gluten-core/src/main/scala/io/glutenproject/expression/BinaryExpressionTransformer.scala

```diff
@@ -72,23 +82,110 @@ class LikeTransformer(substraitExprName: String,
         FunctionConfig.OPT))
 
     // CH backend does not support escapeChar, so skip it here.
-    val expressionNodes = if (BackendsApiManager.getBackendName
-      .equalsIgnoreCase(GlutenConfig.GLUTEN_CLICKHOUSE_BACKEND)) {
-      Lists.newArrayList(leftNode, rightNode)
-    } else {
-      Lists.newArrayList(leftNode, rightNode, escapeCharNode)
-    }
+    val expressionNodes =
+      if (
+        BackendsApiManager.getBackendName
+          .equalsIgnoreCase(GlutenConfig.GLUTEN_CLICKHOUSE_BACKEND)
+      ) {
+        Lists.newArrayList(leftNode, rightNode)
+      } else {
+        Lists.newArrayList(leftNode, rightNode, escapeCharNode)
+      }
     val typeNode = ConverterUtils.getTypeNode(original.dataType, original.nullable)
     ExpressionBuilder.makeScalarFunction(functionId, expressionNodes, typeNode)
   }
 }
 
+abstract class Sha2BaseTransformer(
+    substraitExprName: String,
+    left: ExpressionTransformer,
+    right: ExpressionTransformer,
+    original: Expression)
+  extends ExpressionTransformer
+  with Logging {
+  override def doTransform(args: java.lang.Object): ExpressionNode = {
+    BinaryExpressionTransformer(substraitExprName, left, right, original).doTransform(args)
+  }
+}
+
+case class Sha2Transformer(
+    substraitExprName: String,
+    left: ExpressionTransformer,
+    right: ExpressionTransformer,
+    original: Expression)
+  extends Sha2BaseTransformer(substraitExprName, left, right, original) {}
+
+case class CHSha2Transformer(
```

Comment:
It would be better to move a backend-specific class into that backend's own module, e.g., `backends-clickhouse/src/main/scala/io/glutenproject/expression/`. Right?